pkg/loki/config_wrapper.go

```diff
@@ -147,6 +151,19 @@ func applyIngesterRingConfig(r *ConfigWrapper) {
 	r.QueryScheduler.SchedulerRing.TokensFilePath = lc.TokensFilePath
 	r.QueryScheduler.SchedulerRing.KVStore.Store = s
 	r.QueryScheduler.SchedulerRing.KVStore.StoreConfig = sc
+
+	// Compactor
```

Comment:
The way the dynamic config is applied, the user supplied defaults are applied after we do the dynamic config, so it would replace anything we do. I think it's simpler to do it this way so we have less conditionals.

The storage config conditionals are more to make sure we don't add configs for storage engines not in use.